### PR TITLE
Decorate

### PR DESCRIPTION
**Pull request by V0pe**
### Pattern Decoration

- [ ] Add the base class
- [ ] Add the base decorator
- [ ] Add the pattern decorator

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -1,20 +1,6 @@
...
-      'ConcreteComponent'
-    end
-  end
   class Decorator < Nameable
-    attr_accessor :name
-  
-    # @param [Component] component
     def initialize(nameable)
       @nameable = nameable
     end
```
Comment (Kingstalux) [inline decorator.rb]: - [ ] Good job so far 👍 Kindly add the `attr_accessor :nameable` to your `Decorator` class 😄

File: decorator.rb [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase!
+  end
```
Comment (Kingstalux) [inline decorator.rb]: - [ ] You are to capitalize the first letter of the given name but your method capitalizes the whole name. Kindly rectify this issue 👍

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @V0pe, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

175083599-879bf673-fb93-406f-939c-de7e586bf6d9

## Highlights 👏 
✅ Linters are passing 👌 
✅ Good commit history 👍 
✅ Correct Gitflow used 👏 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator.rb (hunk 1 above)
Comment (V0pe) [reply decorator.rb]: I added but removed because it didnt affect result

File: decorator.rb (hunk 2 above)
Comment (V0pe) [reply decorator.rb]: @Kingstalux my bad. I thought it was uppercase all. Thanks 👍

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @V0pe 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Github inline comments feature is used properly.
- You asked for changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by V0pe** [comment]
@emmanuellekamwa Merci.

**Pull request merged**